src/OrchardCore/OrchardCore.Roles.Core/DefaultOwnerRoleCache.cs

```diff
@@ -0,0 +1,147 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using OrchardCore.Infrastructure.Security;
+using OrchardCore.Security;
+
+namespace OrchardCore.Roles.Core;
+
+public class DefaultOwnerRoleCache : IOwnerRoleCache
```

Comment:
I think you should remove this cache. The DocumentManager is already caching the instance, and knows about mutable/immutable which you don't handle here. Maybe the controller is actually updating immutable objects and could even be a security issue (failed validation but cache entity still updated and shared).

Also the performance gains must be really hard to get since `.Roles` is already a cached instance in memory. The only gain would be the filter on the flag.

Finally, you mentioned it, there is a risk that roles are updated in a separate node and the memory cache won't see it. At least with DocumentManager, which is already in place without this cache, it is refreshed regularly. Though an ideal solution would be a backplane (redis, sql server, ...) to invalidate with push (not polling).

I assume the handler only check for permissions on authenticated users (that limits the impact at least), and could you confirm that with an authenticated user and without this cache, there is a db query on every request? I would hope that the DocumentManager is working fine there to prevent the query.

Reply:
PS: I have looked at HybridCache and other OSS solutions and I think we should discuss our options for a broader change.

Reply:
@sebastienros 

>PS: I have looked at HybridCache and other OSS solutions and I think we should discuss our options for a broader change.

I agree. We should have a service that handles caching and cache busting that we can just utilize "include in the document manager". It would be amazing to have `ICacheService` or other names that would return an instance of an object from memory or distributed. At the same time, this service would have to clear cache from memory from each web service when the memory cache should be cleared.

> I think you should remove this cache. 
I did that. We no longer have `IOwnerRoleCache`. Instead, we use the IRoleService which gets the roles from the memory or the database as needed.

Can you please review this PR?